### ISSUE TITLE
What is the use of user_type in addUserRoleRequest ?

### ISSUE THREAD

**Issue by pdumais** [openconfig/ovgs#8]
I think I understand that the creation of users would be a function of the vendor. So the user type would be determined at that time. So what is the purpose of this field in the AddUserRoleRequest?

**Comment by haussli** [comment]
Yes, users and their authentication are determined externally.  For example, from the same authentication source as support or forum sites.

Certainly an implementation *could* impose the ovgs role, or initial role, for each user from that external source.  Certainly, support staff must have SUPPORT role and at least one user must have ADMIN role.  From there, the ADMIN role can add other users from that same authentication source to any group in a given tree as an ADMIN or lesser role.  Note that not every user from a given customer company should have access AND the user need not be from the same customer company - ie: gear can be owned by one company, but managed by another party.

**Comment by pdumais** [comment]
Is it possible that your answer is refering to user_group? I was inquiring about user_type. I'm wondering why this parameter is required when adding a user role.

**Comment by haussli** [comment]
type is relevant to the external auth source; I am unaware of any use of it within the RPCs.  in some cases the user is a real user, sometimes it is robot.  that robot could be using an auth token associated with a real user but with other constraints.

if you were creating an implementation of the OVGS with an authentication backend that did not support this differentiation, the field could be ignored.
